Add pull request template (#29)
amazeeio/amazeeai-k0rdent-catalog: .github/pull_request_template.md

> ## What does this MR do?

Reviewer: ```suggestion
## What does this PR do?
```

Author: Good point! I forgot to change that

> <!-- Briefly describe what this MR is about -->

Reviewer: ```suggestion
<!-- Briefly describe what this PR is about -->
```

## Related issue(s)

<!-- 
Please link relevant issues to help with tracking.

Examples:
- Fixes #<issue number> or <issue link>
- Closes #<issue number> or <issue link>
- Relates #<issue number> or <issue link>
-->

> ## Related MR(s)

Reviewer: ```suggestion
## Related changes
```

> <!-- Please link any other related MRs with this change. -->

Reviewer: ```suggestion
<!-- Please link any other PRs or MRs related to this change. -->
```

## Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change which adds no functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How has this been tested?

<!-- Please describe the tests that you performed to verify your changes. -->

## How and when this is going to be rolled out?

<!-- Please describe the rollout plan. -->

## Checklist

<!-- Make sure that you’ve checked the boxes below before you add any reviewers in the MR. -->

- [ ] Assign yourself as the Assignee of this MR.
- [ ] Review your changes before adding any reviewers.
- [ ] (If applicable) Documentation is updated (README, Notion etc.)
- [ ] If you have multiple commits, please combine them into a few logically organized commits by squashing them.

> - [ ] This MR contains substantial use of AI-assisted coding tools.

Reviewer: ```suggestion
- [ ] This PR contains substantial use of AI-assisted coding tools.
```

## Status

> <!-- Please also mark the MR as Draft if it isn't ready for review. -->

Reviewer: I think we could just have the comment and not an extra checkbox since this is already tracked via draft/non-draft
```suggestion
<!-- Please also mark the PR as Draft if it isn't ready for review. -->
```
